storage/cloud.py

```diff
@@ -396,6 +396,13 @@ def initiate_chunked_upload(self):
     def stream_upload_chunk(self, uuid, offset, length, in_fp, storage_metadata, content_type=None):
         self._initialize_cloud_conn()
 
+        # Do not create zero-byte objects.
+        # TODO: length=-1 appears to be a special case and used in tests. Why? Document it.
```

Comment:
Length `-1` means to write the whole `in_fp` stream, vs just `length` bytes if set.

Reply:
Good to know. Thank you!